File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        i, j = 0, len(height) - 1
        while i != j:
            a = min(height[i], height[j])
            area = max(area, (j - i) * a)
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return area

File: 11-container-with-most-water/container-with-most-water.c
int maxArea(int* height, int heightSize) {
    int area=0, i=0, j=heightSize-1, h;

    while (i<j){

        if (height[i]<height[j]) h=height[i++];
        else h=height[j--];

        if ((j-i+1)*h>area) area=(j-i+1)*h;
    }
    return area;
}